From sebek64@post.cz  Sun Jan 25 14:29:30 2009
From: Marcel Sebek <sebek64@post.cz>
Date: Sun, 28 Dec 2008 14:06:50 +0100
Subject: USB: 'option' driver - onda device MT503HS has wrong id
To: smurf@smurf.noris.de, gregkh@suse.de, domenico.riccio@gmail.com
Message-ID: <20081228130650.GA5849@penguin.penguin.peng>
Content-Disposition: inline


While trying to make GSM modem Onda MT503HS working, I found a mismatch
between device id in the driver code (0x0200) and id in the lsusb
output (0x2000).

This patch fixed it for me, but I don't know if the original device id was
also correct and the new ID should be added instead of replacing the
old one.


Signed-off-by: Marcel Sebek <sebek64@post.cz>
Acked-by: Domenico Riccio <domenico.riccio@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/option.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/serial/option.c
+++ b/drivers/usb/serial/option.c
@@ -224,7 +224,7 @@ static int  option_send_setup(struct tty
 #define ONDA_VENDOR_ID				0x19d2
 #define ONDA_PRODUCT_MSA501HS			0x0001
 #define ONDA_PRODUCT_ET502HS			0x0002
-#define ONDA_PRODUCT_MT503HS			0x0200
+#define ONDA_PRODUCT_MT503HS			0x2000
 
 #define BANDRICH_VENDOR_ID			0x1A8D
 #define BANDRICH_PRODUCT_C100_1			0x1002
